### ISSUE TITLE
Windows hns loadbalancers disappear intermittently

### ISSUE THREAD

**Issue by zhiweiv** [kubernetes/kubernetes#110849]
Windows hns loadbalancers in Windows node may disappear suddenly, I run **hnsdiag list loadbalancers** every minutes with Host Process Container on each Windows node. When issue happen, the load balancer count dropped to 1 and recover slowly, it took a few days to come back to normal. 

The script for troubleshooting
```
while($true)
{
	$time = Get-Date -Format 'MM-dd HH:mm'
	$count = (hnsdiag list loadbalancers | Measure-Object -Line).Lines
	echo "$time - loadbalancers: $count"
	Start-Sleep 60
}
```
The output I collected before, more logs here [hns.txt](https://github.com/kubernetes/kubernetes/files/9006642/hns.txt)

```
06-06 02:57 - loadbalancers: 577
06-06 02:58 - loadbalancers: 577
06-06 02:59 - loadbalancers: 577
06-06 03:00 - loadbalancers: 1
06-06 03:11 - loadbalancers: 15
06-06 03:12 - loadbalancers: 15
06-06 03:13 - loadbalancers: 15
```

If I restart kubeproxy, the load balancer count will come back to normal in a few minutes.

This issue was first observed at September 2021, across multiple Windows/k8s versions, we are now on AKS 1.23.5

/sig windows
/kind bug

**Comment by zhiweiv** [comment]
@jsturtevant @perosb
We can discuss the issue here.

**Comment by backjo** [comment]
FWIW, we're seeing similar behavior on Amazon EKS Windows Nodes running 1.22.6

**Comment by jsturtevant** [comment]
Thanks for creating a seperate issue on this.  
/triage accepted

> If I restart kubeproxy, the load balancer count will come back to normal in a few minutes.

I think @daschott initially thought this might be related to https://github.com/kubernetes/kubernetes/issues/109162 where syncing proxy rules takes a long time but since you are able to restart and the rules come back in a few minutes this doesn't seem related.  (the fact that it takes a *few minutes* to sync is but dropping them in the first place doesn't seem related).  

Just in case are you able to test on 1.24.2 or use a custom version?  The fix for this was released in that patch version and will be in the next 1.23 path release via https://github.com/kubernetes/kubernetes/pull/110702.

**Comment by jsturtevant** [comment]
A few more questions:

- Are you able to reproduce this at all?  
- Are you aware of any related events that cause this to happen? 
- Something like node restart, or redeploying multiple pods at the same time or something else major changing in the system? 
- Does it happen to only one node or all nodes?

**Comment by AbelHu** [comment]
@zhiweiv 1.24.0 is in preview in AKS and 1.24 will be GA in July. https://docs.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar

**Comment by zhiweiv** [comment]
The issue seems fixed after upgrade to AKS 1.23.12, it doesn't occur again in my testing cluster since October.

**Comment by zhiweiv** [comment]
It has been fixed in AKS.
